src/core/libraries/kernel/memory_management.cpp

```diff
@@ -207,6 +208,35 @@ int PS4_SYSV_ABI sceKernelQueryMemoryProtection(void* addr, void** start, void**
     return memory->QueryProtection(std::bit_cast<VAddr>(addr), start, end, prot);
 }
 
+int PS4_SYSV_ABI sceKernelMProtect(const void* addr, size_t size, int prot) {
+
+    Core::MemoryManager* memory_manager = Core::Memory::Instance();
+
+    // Cast the 'prot' integer to 'MemoryProt' enum type
+    Core::MemoryProt protection_flags = static_cast<Core::MemoryProt>(prot);
+
+    int result = memory_manager->Protect(std::bit_cast<VAddr>(addr), size, protection_flags);
+    if (result != ORBIS_OK) {
+        LOG_ERROR(Kernel_Vmm, "MProtect failed with result {}", result);
+    }
+    return result;
```

Comment:
```suggestion
    Core::MemoryManager* memory_manager = Core::Memory::Instance();
    const auto protection_flags = static_cast<Core::MemoryProt>(prot);
    return memory_manager->Protect(std::bit_cast<VAddr>(addr), size, protection_flags);
```
The logging can be done in Protect()